### PR TITLE
feat: add shareable framework configs

### PR DESCRIPTION
**Pull request by Belco90**
This PR adds shareable framework configs (angular, react, vue) and updates corresponding documentation.

Apart from the setup itself, I've enabled `no-debug` rule for all these 3 frameworks included.

Additionally, I've added some snapshots tests for exposed configs so we have a way to test if they are changed or not flexible enough to be updated easily when adding/modifying rules. Related to this, I added couple of eslint plugins for jest as this is the first test file purely written in jest syntax.

### CODE REVIEW

File: tests/index.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+'use strict';
```
Comment (thomlom) [inline tests/index.js]: What about looping over the configurations to make the test lighter?

```js
'use strict';

const { configs } = require('../lib');

describe('shareable configs', () => {
  Object.entries(configs).forEach(([configName, config]) => {
    it(`should export proper ${configName} config`, () => {
      expect(config).toMatchSnapshot()
    })
  })
});

```

Comment (Belco90) [reply tests/index.js]: Even easier than that is to use [test.each](https://jestjs.io/docs/en/api#testeachtable-name-fn-timeout) so we have one definition with different values. I just add them like this to specify proper configs with snapshots, but I'm happy to implement these tests like that.

**Comment by Belco90** [comment]
>     * I'm not sure if snapshots are necessary or not. At least, it's a way to "lock" the configuration and a way to warn us if we modify it.
> 
>     * As for jest formatting, I'm wondering what is its value? Isn't prettier already formatting the JS files?

Exactly, snapshots it's best way to lock the config and easiest to update. About jest formatting, it doesn't format jest syntax and eslint doesn't recognize jest methods (describe, test, it, etc) so I've installed it also for that reason.

File: tests/index.js (hunk 1 above)
Comment (thomlom) [reply tests/index.js]: Yes, why not!

**Comment by thomlom** [comment]
A simpler way to make ESLint recognize Jest methods is to add: 

```
{
  "env": {
    "jest": true
  }
}
```

to `.eslintrc.json`. But in the end, it does no harm to keep this Jest formatting plugin. As you wish!

**Comment by Belco90** [comment]
Tests updated with `test.each` to avoid duplication.

**Comment by thomlom** [comment]
Great!

**Pull request merged**